src/main/java/net/javadiscord/javabot/data/h2db/DatabaseRepository.java

```diff
@@ -0,0 +1,183 @@
+	 * model class. Additionally, this value is then wrapped in an {@link Optional} as it is possible for the value
+	 * to be empty.
+	 *
+	 * @param filter The filter for this query. Might look something like "WHEN age > 18".
+	 * @param args   Additional arguments used for formatting. "?" symbols are used as placeholders.
+	 * @return An {@link Optional} which eventually holds the desired value.
+	 * @throws SQLException If an error occurs.
+	 */
+	public final Optional<T> querySingle(String filter, @NotNull Object... args) throws SQLException {
+		try (PreparedStatement stmt = con.prepareStatement(String.format("SELECT * FROM %s%s", tableName, filter.isEmpty() ? "" : " " + filter))) {
+			int i = 1;
+			for (Object arg : args) {
+				stmt.setObject(i++, arg);
+			}
+			ResultSet rs = stmt.executeQuery();
+			T t = null;
+			if (rs.next()) {
+				t = read(rs);
+			}
+			return Optional.ofNullable(t);
+		}
+	}
+
+	/**
+	 * Queries a multiple rows which match the given filter and converts them to the specified
+	 * model class.
+	 *
+	 * @param filter The filter for this query. Might look something like "WHEN age > 18".
+	 * @param args   Additional arguments used for formatting. "?" symbols are used as placeholders.
+	 * @return An unmodifiable {@link List} which holds the desired value(s).
+	 * @throws SQLException If an error occurs.
+	 */
+	public final @NotNull List<T> queryMultiple(String filter, Object @NotNull ... args) throws SQLException {
+		try (PreparedStatement stmt = con.prepareStatement(String.format("SELECT * FROM %s%s", tableName, filter.isEmpty() ? "" : " " + filter))) {
+			int i = 1;
+			for (Object arg : args) {
+				stmt.setObject(i++, arg);
+			}
+			ResultSet rs = stmt.executeQuery();
+			List<T> list = new ArrayList<>();
+			while (rs.next()) {
+				list.add(read(rs));
+			}
+			return list;
+		}
+	}
+
+	public final long count() {
+		return DbActions.count("SELECT COUNT (*) FROM " + tableName);
+	}
+
+	public final long count(String query, Object... args) {
+		return DbActions.count(String.format(query, args));
+	}
+
+	public final int getLogicalSize() {
+		return DbActions.getLogicalSize(tableName);
+	}
+
+	/**
+	 * Reads the given {@link ResultSet} and tries to convert it to the specified model class {@link T}.
+	 *
+	 * @param rs The {@link ResultSet} to read.
+	 * @return The specified model class {@link T}.
+	 * @throws SQLException If an error occurs.
+	 */
+	public final T read(ResultSet rs) throws SQLException {
+		T instance = instantiate();
+		for (TableProperty<T> property : properties) {
+			property.getConsumer().accept(instance, readResultSetValue(property, rs));
+		}
+		return instance;
+	}
+
+	public final Connection getConnection() {
+		return con;
+	}
+
+	/**
+	 * Closes the {@link Connection}, if not already done.
+	 *
+	 * @throws SQLException If an error occurs.
+	 */
+	public void close() throws SQLException {
```

Comment:
Implement `AutoClosable` and add `@Override` to the `close()` method.

Reply:
Oh. nevermind. This is just a left-over. I'll remove this. All methods use try-with-resources so this is not needed anyway